asv_bench/benchmarks/benchmarks.py

```diff
@@ -1184,4 +1184,96 @@ def time_series_indices(self, shape):
         self.series.groupby(self.series).indices
 
 
+class TimeRepr:
+
+    params = [get_benchmark_shapes("TimeRepr")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        self.df = IMPL.DataFrame(np.random.randn(*shape))
+        execute(self.df)
+
+    # returns a  string thus not calling execute
+    def time_repr(self, shape):
+        repr(self.df)
+
+
+class TimeMaskBool:
+
+    params = [get_benchmark_shapes("TimeMaskBool")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        df = IMPL.DataFrame(np.random.randn(*shape))
+        df = df.where(df > 0)
+        self.bools = df > 0
+        self.mask = IMPL.isnull(df)
+        execute(self.bools), execute(self.mask)
+
+    def time_frame_mask_bools(self, shape):
+        execute(self.bools.mask(self.mask))
+
+    def time_frame_mask_floats(self, shape):
+        execute(self.bools.astype(float).mask(self.mask))
+
+
+class TimeIsnull:
+
+    params = [get_benchmark_shapes("TimeIsnull")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        self.df_no_null = IMPL.DataFrame(np.random.randn(*shape))
+        sample = np.array([np.nan, 1.0])
+        data = np.random.choice(sample, (shape[0], shape[1]))
+        self.df = IMPL.DataFrame(data)
+        execute(self.df), execute(self.df_no_null)
+
+    def time_isnull_floats_no_null(self, shape):
+        execute(IMPL.isnull(self.df_no_null))
+
+    def time_isnull(self, shape):
+        execute(IMPL.isnull(self.df))
+
+
+class TimeDropna:
+
+    params = (["all", "any"], [0, 1], get_benchmark_shapes("TimeDropna"))
+    param_names = ["how", "axis", "shape"]
+
+    def setup(self, how, axis, shape):
+        row, col = shape
+        self.df = IMPL.DataFrame(np.random.randn(row, col))
+        self.df.iloc[row // 20 : row // 10, col // 3 : col // 2] = np.nan
+        self.df_mixed = self.df.copy()
+        self.df_mixed["foo"] = "bar"
+        execute(self.df), execute(self.df_mixed)
+
+    def time_dropna(self, how, axis, shape):
+        execute(self.df.dropna(how=how, axis=axis))
+
+    def time_dropna_axis_mixed_dtypes(self, how, axis, shape):
+        execute(self.df_mixed.dropna(how=how, axis=axis))
+
+
+class TimeEquals:
+
+    params = [get_benchmark_shapes("TimeEquals")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        self.df = IMPL.DataFrame(np.random.randn(*shape))
+        self.df_nan = self.df.copy()
+        self.df_nan.iloc[-1, -1] = np.nan
+        execute(self.df), execute(self.df_nan)
+
+    # returns a  boolean thus not calling execute
+    def time_frame_float_equal(self, shape):
+        self.df.equals(self.df)
+
+    # returns a  boolean thus not calling execute
+    def time_frame_float_unequal(self, shape):
+        self.df.equals(self.df_nan)
```

Comment:
Let's drop one case for Modin. Having another case is enough to track performance for `equals`.

```suggestion
    def setup(self, shape):
        self.df = IMPL.DataFrame(np.random.randn(*shape))
        self.df.iloc[-1, -1] = np.nan
        execute(self.df)

    # returns a boolean thus not calling execute
    def time_frame_float_unequal(self, shape):
        self.df.equals(self.df_nan)
```